### PR TITLE
add languageCode type

### PR DESCRIPTION
**Pull request by carafelix**
Would be helpful for getting autocompletion in the commands plugin while localizing commands, could potentially improve readability in other places of the codebase.

```ts
userCommands.command("start", "example", () => {})
            .localize('<AUTOCOMPLETES>')
```
reference:
https://www.loc.gov/standards/iso639-2/php/code_list.php 
https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes 
https://core.telegram.org/bots/api#setmycommands <-- spec in telegram

### CODE REVIEW

File: langs.ts
Comment (roj1512) [inline langs.ts:1]: Some languages have only 3-letter ISO codes.

Comment (KnorpelSenf) [reply langs.ts:1]: They are not supported anyway. 
> A two-letter ISO 639-1 language code. If empty, commands will be applied to all users from the given scope, for whose language there are no dedicated commands

Comment (roj1512) [reply langs.ts:1]: Telegram translations support them tho. I should give it a try anyway.

Comment (roj1512) [reply langs.ts:1]: Yeah, they are not supported.

Comment (carafelix) [reply langs.ts:1]: So we are good to go?

**Comment by carafelix** [comment]
Should I declared this just locally for the commands plugin or better intended here?

File: langs.ts
Comment (KnorpelSenf) [reply langs.ts:1]: Yes

**Review by KnorpelSenf — CHANGES_REQUESTED** [review]
I like
- that we are able to narrow down the types this way, it is specified well enough that we can do this
- that you added the types to this repo

I do not like that you used a TS enum, they are basically not much more than a footgun and since we do not expose any of its functionality, we might as well directly go with
```ts
export type LanguageCode =
  | "ab" // Abkhazian
  | "aa" // Afar
  // ...
```
Could you change this?

**Comment by roziscoding** [comment]
> I do not like that you used a TS enum, they are basically not much more than a footgun and since we do not expose any of its functionality, we might as well directly go with
> 
> ```ts
> export type LanguageCode =
>   | "ab" // Abkhazian
>   | "aa" // Afar
>   // ...
> ```

I like the idea of not going with an enum, but I think having these at runtime would be useful so users don't need to look for the correct code for the language they're using, so I'd suggest we go with

```ts
export const LanguageCodes = {
  "ab": "Abkhazian",
  // ...
}

export type LanguageCode = keyof typeof LanguageCodes
```

**Comment by roziscoding** [comment]
> I do not like that you used a TS enum, they are basically not much more than a footgun and since we do not expose any of its functionality, we might as well directly go with
> 
> ```ts
> export type LanguageCode =
>   | "ab" // Abkhazian
>   | "aa" // Afar
>   // ...
> ```

I like the idea of not going with an enum, but I think having these at runtime would be useful so users don't need to look for the correct code for the language they're using, so I'd suggest we go with

```ts
const LanguageCodes = {
  "Abkhazian": "ab",
  "Afar": "aa",
  // ...
} as const;

export type LanguageCode = typeof LanguageCodes[keyof typeof LanguageCodes];
```

**Comment by carafelix** [comment]
> I think having these at runtime would be useful so users don't need to look for the correct code for the language they're using.

Yeah I was thinking exactly that. I did the same thing with an object in a gist. Could any of you explain why is not proper to use an enum for it?

**Comment by KnorpelSenf** [comment]
> I think having these at runtime would be useful

So far, it's been a strict rule that this package exposes types only, and 0 bytes of executable code. We only really use `.ts` files because `.d.ts` was hard to import on Deno back then.

It's mostly for historic reasons that I'm not comfortable with this, but historic reasons are bad reasons, so please give me a few days to change my mind or to come up with a better solution :D

**Comment by KnorpelSenf** [comment]
> Could any of you explain why is not proper to use an enum for it?

There are many articles on the web that explain this better than i can could in a GitHub comment. One of them is https://dev.to/ivanzm123/dont-use-enums-in-typescript-they-are-very-dangerous-57bh.

File: langs.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
+/** A two-letter ISO 639-1 language code.
+ * @see https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes
+ * @see https://www.loc.gov/standards/iso639-2/php/code_list.php
+ */
+export type LanguageCode = typeof LanguageCodes[keyof typeof LanguageCodes];
+
+/**
+ * @see {@link LanguageCode}
+ */
+const LanguageCodes = {
```
Comment (KnorpelSenf) [inline langs.ts:10]:
```suggestion
export const LanguageCodes = {
```
because otherwise there's no reason to have this value at runtime at all

**Comment by carafelix** [comment]
Examples of the readability benefiting from adding the Language Code runtime object. It's more clear for languages that their respective code does not resemble their name in english :
```ts
const i18n = new I18n<MyContext>({
  defaultLocale: LanguageCodes.German, // Instead of 'de'
  directory: "locales",
});
i18n.loadLocaleSync("LanguageCodes.Greek", { // Instead of 'el'
  source: `greeting = Γειά σου { $name }!
language-set = Λορεμ ιπσθμ δολορ σιτ αμετ`,
});
```
```ts
cmds.command("duke", "_", () => {})
                .localize(LanguageCodes.Croatian, "vojvoda", "_") // instead of 'hr'
```                
```ts
bot.hears('something', (ctx) => {
  ctx.api.setMyShortDescription('快乐机器人', LanguageCodes.Chinese) // instead of 'zh'
  ctx.api.setMyName('robot sásta', LanguageCodes.Irish) // instead of 'ga'
})
```

**Review by KnorpelSenf — APPROVED** [review]

**Pull request merged**